spec-tests/src/test/java/org/candlepin/spec/PoolSpecTest.java

```diff
@@ -0,0 +1,673 @@
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertNotFound;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.candlepin.ApiException;
+import org.candlepin.dto.api.v1.AttributeDTO;
+import org.candlepin.dto.api.v1.BrandingDTO;
+import org.candlepin.dto.api.v1.ConsumerDTO;
+import org.candlepin.dto.api.v1.ConsumerTypeDTO;
+import org.candlepin.dto.api.v1.EntitlementDTO;
+import org.candlepin.dto.api.v1.OwnerDTO;
+import org.candlepin.dto.api.v1.PoolDTO;
+import org.candlepin.dto.api.v1.ProductDTO;
+import org.candlepin.dto.api.v1.UserDTO;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+
+
+
+@SpecTest
+class PoolSpecTest {
+
+    /**
+     * Generates a random suffix of 7 characters or fewer. Alternative to StringUtil.random which
+     * generates too many characters for fields which have small length limitations.
+     *
+     * @param prefix
+     *  The base string to concatenate to the generated suffix
+     *
+     * @return
+     *  the provided base string/prefix concatenated with a randomly generated suffix
+     */
+    private String randomSuffix(String prefix) {
+        return prefix + "-" + ((int) (Math.random() * 100000));
+    }
+
+    private ApiClient createUserClient(ApiClient client, OwnerDTO owner) throws ApiException {
+        UserDTO user = UserUtil.createUser(client, owner);
+        return ApiClients.trustedUser(user.getUsername(), true);
+    }
+
+    private ApiClient createConsumerClient(ApiClient client, OwnerDTO owner) throws ApiException {
+        String consumerName = this.randomSuffix("test_consumer");
+        ConsumerDTO consumer = this.createConsumer(client, owner, consumerName, "system", null);
+
+        return this.createConsumerClient(client, consumer);
+    }
+
+    private ApiClient createConsumerClient(ApiClient client, ConsumerDTO consumer) throws ApiException {
+        return ApiClients.ssl(consumer.getIdCert());
+    }
+
+    private OwnerDTO createOwner(ApiClient client) throws ApiException {
+        return client.owners().createOwner(Owners.random());
+    }
+
+    private ConsumerDTO createConsumer(ApiClient client, OwnerDTO owner, String name, String type,
+        Map<String, String> facts) throws ApiException {
+
+        ConsumerTypeDTO ctype = new ConsumerTypeDTO()
+            .label(type);
+
+        ConsumerDTO consumer = new ConsumerDTO()
```

Comment:
Check ConsumerTypes/Consumer.random

Reply:
In both the previous version and current version of the consumers factory, the usage is slightly clunky without changing potentially every other use of it out there. Specifically, that NestedOwnerDTO and OwnerDTO aren't at all related makes it real annoying to make the appropriate factory functions that cover our use cases (again, without clobbering other tests).

I've started on this a couple times now, and in both cases, I felt that leaving the manual construction will be cleaner.

ConsumerTypes looks to suffer the same problem as ProductAttributes in that it's more work to build/unwrap the DTO from the enum than it is to just build the DTO.